app/controllers/concerns/view_component/preview_actions.rb

```diff
@@ -59,7 +59,7 @@ def show_previews?
     def find_preview
       candidates = []
       params[:path].to_s.scan(%r{/|$}) { candidates << Regexp.last_match.pre_match }
-      preview = candidates.detect { |candidate| ViewComponent::Preview.exists?(candidate) }
+      preview = candidates.sort_by { |candidate| -candidate.length }.detect { |candidate| ViewComponent::Preview.exists?(candidate) }
```

Comment:
```suggestion
      preview = candidates.sort_by(&:length).reverse_each.detect { |candidate| ViewComponent::Preview.exists?(candidate) }
```

May be a bit clearer?